packages/core/cache/src/Cache.js

```diff
@@ -49,8 +50,8 @@ export default class Cache {
     this.dirExists = true;
   }
 
-  getCacheId(appendedData: string) {
-    return md5(this.optionsHash + appendedData);
+  getCacheId(appendedData: string, env: Environment) {
```

Comment:
I think it would be nice to have a generic implementation of the cache that is agnostic as to what is being cached, and some abstraction on top of it that is specific to what we are caching. Kind of like Graph and AssetGraph. That way we can have a well tested base that shouldn't change much, and the abstraction on top can be thinner and easier to understand when we change how things are cached. It's not something that needs to be done as part of this PR, just something I've been thinking about.

Reply:
Yeah that's a nice idea. Would they both live in the `@parcel/cache` package though?